python/dlisio/plumbing/frame.py

```diff
@@ -118,6 +118,32 @@ def __init__(self, obj = None, name = None, file = None):
         # Defaults to Frame.dtype_format
         self.dtype_fmt = self.dtype_format
 
+    @property
+    def index(self):
```

Comment:
What's misleading exactly, and for who? what is index-on-its-own in the frame/channel case? 

It seams like Frames without index-channels are rare, which makes perfect sense. As you say, indexing by frameno is pretty meaningless. 

Having it as a method yield some weird syntax that i dont like, e.g. `frame.index().curves`, so I prefer having it as a property

Reply:
Then lets say we have one of these rare cases (I will trust whatever statistics you have for stating that this is rara) where there is no index channel.

I am just a user, I have no idea about how this indexing works. I call  frame.index, because this is what I always do. I receive a warning and None as the result. Now I am confused. Previous 99 frames had index and this one doesn't? What does that supposed to mean? How can there be no index? There is nothing to clarify my confusion.

So as you showed that putting into in pandas dataframe will still be fine (user again will get just a warning ("What did I do now??? I was creating pandas dataframe, I thought it went fine! ")), then at least some documentation would be appreciated.

If you don't like this syntax so much - sure. To me all is kinda same.

Reply:
help(frame.index) gives you the answer you seek